lib/aasm/persistence/base.rb

```diff
@@ -84,6 +84,11 @@ def state_with_scope(name, *args)
     end
     alias_method :state_without_scope, :state
     alias_method :state, :state_with_scope
+
+  private
+    def should_create_scope?(name, *args)
+      return @state_machine.config.create_scopes && !@klass.respond_to?(name) && (args.first.blank? || args.first[:create_scope] != false)
```

Comment:
You'd think so! But it looks like the problem is deep inside arel it checks if the object responds to `locked`, and if it does, it calls it on the arel object, which for some reason is a different object. So if a scope named `locked` exists, it fails.